### PR TITLE
Pin framework to tf in AIR rl offline trainer example

### PR DESCRIPTION
**Pull request by avnishn**
# something broken about the way that torch ppo writes out to json

when using the output writer, torch ppo writes out sample batches in the incorrect format:

`“value”: {”type”: “MultiAgentBatch”...`

instead of just

`{”type”: “MultiAgentBatch”...`

Useful debug script:
[https://gist.github.com/avnishn/c8599a9947edcd9934421713ed82f175](https://gist.github.com/avnishn/c8599a9947edcd9934421713ed82f175)

```python
def generate_offline_data(path: str):
    print(f"Generating offline data for training at {path}")
    trainer = RLTrainer(
        algorithm="PPO",
        run_config=RunConfig(stop={"timesteps_total": 5000}),
        config={
            "env": "CartPole-v1",
            "output": "dataset",
            "output_config": {
                "format": "json",
                "path": path,
                "max_num_samples_per_file": 1,
            },
            "batch_mode": "complete_episodes",
            "framework": "tf"
        },
    )
    trainer.fit()
```

```python
def train_rl_bc_offline(path: str, num_workers: int, use_gpu: bool = False) -> Result:
    print("Starting offline training")
    dataset = ray.data.read_json(
        path, parallelism=num_workers, ray_remote_args={"num_cpus": 1}
    )

    trainer = RLTrainer(
        run_config=RunConfig(stop={"training_iteration": 5}),
        scaling_config=ScalingConfig(num_workers=num_workers, use_gpu=use_gpu),
        datasets={"train": dataset},
        algorithm=BC,
        config={
            "env": "CartPole-v1",
            "framework": "tf",
            "evaluation_num_workers": 1,
            "evaluation_interval": 1,
            "evaluation_config": {"input": "sampler"},
            "framework": "tf"
        },
    )

    # Todo (krfricke/xwjiang): Enable checkpoint config in RunConfig
    # result = trainer.fit()
    tuner = Tuner(
        trainer,
        _tuner_kwargs={"checkpoint_at_end": True},
    )
    result = tuner.fit()[0]
    return result
```

```python
ray.init(num_cpus=8)

path = "/tmp/out"
generate_offline_data(path)
```

```python
result = train_rl_bc_offline(path=path, num_workers=2, use_gpu=False)
```

Should work with torch but doesn’t because of the way that sample batches are created.

Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
corresponding issue:
https://github.com/ray-project/ray/issues/33751

**Comment by amogkam** [comment]
can we fix the underlying issue instead?

**Review by Yard1 — APPROVED** [review]

**Comment by avnishn** [comment]
coming in a separate pr @amogkam. I can't fix it before the release cut.

**Comment by amogkam** [comment]
@gjoliver @matthewdeng please review

**Review by kouroshHakha — APPROVED** [review]

File: doc/source/ray-air/examples/rl_offline_example.ipynb [hunk 1]
```diff
@@ -185,233 +187,10 @@
...
-   "execution_count": 6,
+   "execution_count": null,
    "id": "cae1337e",
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "2022-09-26 18:22:15,032\tINFO worker.py:1509 -- Started a local Ray instance. View the dashboard at \u001b[1m\u001b[32m127.0.0.1:8265 \u001b[39m\u001b[22m\n"
```
Comment (matthewdeng) [inline doc/source/ray-air/examples/rl_offline_example.ipynb:196]: We should keep the output?

Comment (avnishn) [reply doc/source/ray-air/examples/rl_offline_example.ipynb:196]: sure thing

**Comment by gjoliver** [comment]
wait the branch cut has already passed, do we still need this?

**Comment by avnishn** [comment]
branch cut is tonight at midnight. I was asked to make this fix since this test was red on master but I'll defer to you guys instead to determine whether or not this needs to be merged.

File: doc/source/ray-air/examples/rl_offline_example.ipynb (hunk 1 above)
Comment (avnishn) [reply doc/source/ray-air/examples/rl_offline_example.ipynb:196]: ok all done.

**Pull request merged**